bitcoin/src/parse.rs

```diff
@@ -110,6 +110,16 @@ pub(crate) fn hex_u32<S: AsRef<str> + Into<String>>(s: S) -> Result<u32, ParseIn
     })
 }
 
+pub(crate) fn hex_u128(s: &str) -> Result<u128, ParseIntError> {
+    u128::from_str_radix(s, 16).map_err(|error| ParseIntError {
+        input: s.into(),
+        bits: u8::try_from(core::mem::size_of::<u128>() * 8).expect("max is 128 bits for u128"),
+        // TODO(Tobin): Copied from above, I don't understand why this is not just `false`?
+        is_signed: u32::try_from(-1i8).is_ok(),
+        source: error,
+    })
+}
```

Comment:
TODO to be removed before merge, anyone know the answer to this?

Reply:
I don't think it does need to be generic in `hex_u32` - it refers to the signed-ness of the integer being parsed, right? So for a `u32` it is plain old `false`. (Fixed in #2577.)